app/forms/state_file/md_two_income_subtractions_form.rb

```diff
@@ -0,0 +1,24 @@
+module StateFile
+  class MdTwoIncomeSubtractionsForm < QuestionsForm
+    set_attributes_for :intake, :primary_student_loan_interest_ded_amount, :spouse_student_loan_interest_ded_amount
+
+    validate :valid_amounts
+
+    def save
+      @intake.update(attributes_for(:intake))
+    end
+
+    private
+
+    def valid_amounts
+      if @intake.direct_file_data.fed_student_loan_interest.present?
+        if (primary_student_loan_interest_ded_amount.to_i + spouse_student_loan_interest_ded_amount.to_i) != @intake.direct_file_data.fed_student_loan_interest.to_i
```

Comment:
Hmm I see a potential problem here:

primary: 20.75 + spouse: 20.75 = total loan interest: 41.50 
use `.to_i` for each value: 
20 + 20 = 41

therefore the sum will never be equal. I guess `fed_student_loan_interest` is always an integer (according to the xsd) so maybe we'll never be in this situation?

I think we should sum the value without converting `to_i`, then convert to an integer (preferably round?) and make sure they're equal? what do you think?

```
(primary_student_loan_interest_ded_amount + spouse_student_loan_interest_ded_amount).round != @intake.direct_file_data.fed_student_loan_interest
```

Reply:
also could we add a spec for this validation?

Reply:
Since the input comes in as a string, I think it's worthwhile to convert both to floats first, then convert the sum to an integer for comparison.

Also yes, added specs for this form including validation